examples/derived_gauges/derived_gauge.go

```diff
@@ -0,0 +1,237 @@
+// Copyright 2019, OpenCensus Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Command stats implements the stats Quick Start example from:
+//   https://opencensus.io/quickstart/go/metrics/
+// START entire
+package main
+
+import (
+	"fmt"
+	"log"
+	"math/rand"
+	"sync"
+	"time"
+
+	"bufio"
+	"go.opencensus.io/exporter/prometheus"
+	"go.opencensus.io/metric"
+	"go.opencensus.io/metric/metricdata"
+	"go.opencensus.io/metric/metricproducer"
+	"net/http"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// This example demonstrates the use of derived gauges. It is a simple interactive program of consumer
+// and producer. User can input number of items to produce. Producer produces specified number of
+// items. Consumer consumes randomly consumes 1-5 items in each attempt. It then sleeps randomly
+// between 1-10 seconds before the next attempt.
+//
+// There are two metrics collected to monitor the queue.
+// 1. queue_size: It is an instantaneous queue size represented using derived gauge int64.
+// 2. queue_seconds_since_processed_last: It is the time elaspsed in seconds since the last time
+//    when the queue was consumed. It is represented using derived gauge float64.
+type queue struct {
+	size         int
+	q            []int
+	lastConsumed time.Time
+	mu           sync.Mutex
+}
+
+var q = &queue{}
+
+const (
+	maxItemsToConsumePerAttempt = 25
+)
+
+func init() {
+	q.q = make([]int, 100)
+}
+
+// consume randomly dequeues upto 5 items from the queue
+func (q *queue) consume() {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	consumeCount := rand.Int() % maxItemsToConsumePerAttempt
+	i := 0
+	for i = 0; i < consumeCount; i++ {
+		if q.size > 0 {
+			q.q = q.q[1:]
+			q.size--
+		} else {
+			break
+		}
+	}
+	if i > 0 {
+		q.lastConsumed = time.Now()
+	}
+}
+
+// produce randomly enqueues upto 5 items from the queue
+func (q *queue) produce(count int) {
+	q.mu.Lock()
+	defer q.mu.Unlock()
+
+	for i := 0; i < count; i++ {
+		v := rand.Int() % 100
+		q.q = append(q.q, v)
+		q.size++
+	}
+	fmt.Printf("queued %d items, queue size is %d\n", count, q.size)
+}
+
+func (q *queue) runConsumer(interval int, cQuit chan bool) {
```

Comment:
s/cQuit/quit

quit is a common variable name for closing channels.

Reply:
done